src/app/modules/admin-core/services/users-service/users.service.ts

```diff
@@ -26,52 +24,36 @@ export class UsersService {
   }
 
   createUser(data) {
+    console.log('!!!!!!!!!!!!!!!', data);
     return this.Http.post(environment.base_url + '/admin-service/api/v1/user-creation/create', data)
   }
 
+  getCurrentUserRoles() {
+    let promise = new Promise((resolve, reject) => {
+      this.Http.get(environment.base_url + '/admin-service/api/v1/platform-user-roles/getProfile')
+        .toPromise().then(
+          res => { // Success
+            this.roles = res['result']['roles'];
+            resolve();
+          }
+        );
+    });
+    return promise;
+  }
 
-  toFormGroup(questions: QuestionBase<string>[]) {
+  toFormGroup(questions) {
     let group: any = {};
-
     questions.forEach(question => {
-      group[question.key] = question.required ? new FormControl(question.value || '', Validators.required)
-        : new FormControl(question.value || '');
+      group[question.field] = 
+      // new FormControl(true);
+      question.visible ? new FormControl(question.value || '', Validators.required): new FormControl(question.value || '');
```

Comment:
why are you not getting validation from the question object?

Reply:
Removed